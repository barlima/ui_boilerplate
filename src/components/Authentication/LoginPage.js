import React from 'react';
import { connect } from 'react-redux';
import axios from 'axios';
import { ToastContainer, toast, Slide } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';

import { login } from '../../actions/auth';
import { Link } from 'react-router-dom';


export class LoginPage extends React.Component {
  state = {
    errors: []
  }

  showErrors = () => {
    if (this.state.errors.length > 0) { 
      return this.state.errors.map((e, index) => (
        toast.error(e, {
          position: toast.POSITION.TOP_LEFT
        })
      ))
    }
  }

  handleLogin = (e) => {
    e.preventDefault();

    const email = e.target.email.value;
    const password = e.target.password.value;
    const request = {"auth": {"email": email, "password": password}};

    axios.post('http://localhost:3001/api/v1/user_token', request)
      .then((res) => {
        const token = res.data.jwt;
        localStorage.setItem("jwt", token);
        this.props.login(token);
      })
      .then(() => {
        this.props.history.push('/dashboard');
      })
      .catch(() => {
        this.setState(() => ({
          errors: ["Invalid username or password"]
        }))
      });
  }

  componentDidUpdate() {
    this.showErrors()
  }

  render() {
    return (
      <div className="box-layout">
        <div className="box-layout__box" >
          <h1 className="box-layout__title">LOGIN</h1>
          
          <ToastContainer transition={Slide} hideProgressBar={true} />

          <form onSubmit={this.handleLogin}>
            <label htmlFor="email">Email: </label>
            <input
              name="email"
              id="email"
              type="email"
            />
            <label htmlFor="password">Password:</label>
            <input
              name="password"
              id="password"
              type="password"
            />
            <br />
            <input 
              name="submit"
              id="submit"
              type="submit"
              className="button"
            />
          </form>
          <p>
            {"New user? "}
            <Link to="/signup">
              Sign Up
            </Link>!
          </p>
        </div>
      </div>
    )
  };
};

const mapDispatchToProps = (dispatch) => ({
  login: (token) => dispatch(login(token))
});

export default connect(undefined, mapDispatchToProps)(LoginPage);