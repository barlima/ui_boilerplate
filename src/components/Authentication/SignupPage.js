import React from 'react';
import axios from 'axios';
import { Link } from 'react-router-dom';

export class SignupPage extends React.Component {
  state = {
    errors: []
  }

  handleSignup = (e) => {
    e.preventDefault();

    const email = e.target.email.value;
    const password = e.target.password.value;
    const passwordConfirmation = e.target.passwordConfirmation.value;

    if (password === passwordConfirmation) {
      const request = {
        "user": {
          "email": email, 
          "password": password
        }
      };

      axios.post('http://localhost:3001/api/v1/users/create', request)
        .then((res) => {
          // Show message!
          this.setState(() => ({
            errors: []
          }));

          return res.data;
        })
        .then((data) => {
          if(data.status === 200) {
            this.props.history.push('/login');
          } else {
            console.log(data)
            this.setState((prevState) => ({
                errors: prevState.errors.concat(data.msg)
            }))
          }
        })
        .catch((res) => {
          console.log("POST call failed!")
          console.log(res);
        });

    } else {
      this.setState(() => ({
        errors: ['Password doesn\'t match']
      }));
    }
  }

  render() {
    return (
      <div className="box-layout">
        <div className="box-layout__box" >
          <h1 className="box-layout__title">SIGNUP</h1>
          
          {
            this.state.errors.length > 0 && this.state.errors.map((e, index) => (
              <p key={index}>{e}</p>
            ))
          }

          <form onSubmit={this.handleSignup}>
            <label htmlFor="email">Email: </label>
            <input
              name="email"
              id="email"
              type="email"
            />
            <label htmlFor="password">Password:</label>
            <input
              name="password"
              id="password"
              type="password"
            />
            <label htmlFor="password">Confirm Password:</label>
            <input
              name="passwordConfirmation"
              id="passwordConfirmation"
              type="password"
            />
            <input 
              name="submit"
              id="submit"
              type="submit"
              className="button"
              value="Sign Up"
            />
          </form>
          <p>
            {"or "}
            <Link to="/login">
              Log in
            </Link>!
          </p>
        </div>
      </div>
    )
  }
};

export default SignupPage;