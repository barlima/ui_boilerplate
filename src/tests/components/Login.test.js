import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/Authentication/LoginPage';

test('should render LoginPage', () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});