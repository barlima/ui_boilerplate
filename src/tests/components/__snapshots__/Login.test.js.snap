// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      LOGIN
    </h1>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="email"
      >
        Email: 
      </label>
      <input
        id="email"
        name="email"
        type="email"
      />
      <label
        htmlFor="password"
      >
        Password:
      </label>
      <input
        id="password"
        name="password"
        type="password"
      />
      <br />
      <input
        className="button"
        id="submit"
        name="submit"
        type="submit"
      />
    </form>
    <p>
      New user? 
      <Link
        replace={false}
        to="/signup"
      >
        Sign Up
      </Link>
      !
    </p>
  </div>
</div>
`;
